src/dbus_proxy/_proxy.lua

```diff
@@ -147,11 +147,14 @@ end
 --
 local function call(proxy, interface, method, args)
   local params = build_params(args)
-  local out = proxy._proxy:call_sync(
+  local out,err = proxy._proxy:call_sync(
```

Comment:
In the `lgi` documentation that you link, it says

> if failure occurs, following return values represent error message and error code.

I guess that in this case we'd want to check `out, code, err` here (and then return them)?

Reply:
call_sync doesn't return "code" as additional parameter, only GError as userdata. 

An example:
``` Lua
local res, err = set:Update(someData);
if not res and err then
    print("err:", err," code:", err.code)
    return
end
```

In this case (NetworkManager Update method), the returned error is some type of userdata object which contains a "code" property. When the error is used as string, it just contains the error message, e.g.:
```
GDBus.Error:org.freedesktop.NetworkManager.Settings.Connection.InvalidProperty: ipv4.gateway: gateway cannot be set if there are no addresses configured
```
If the err.code is accessed, it contains the error code:
```
DBUS_ERROR
```

When using other Lua libraries, I see that idiom really often, as described in PIL (https://www.lua.org/pil/8.4.html):
```
The pcall function calls its first argument in protected mode, so that it catches any errors while the function is running. If there are no errors, pcall returns true, plus any values returned by the call. Otherwise, it returns false, plus the error message.

Despite its name, the error message does not have to be a string. Any Lua value that you pass to error will be returned by pcall:

    local status, err = pcall(function () error({code=121}) end)
    print(err.code)  -->  121
```

assert (
https://www.lua.org/pil/8.3.html) in Lua can directly be used with such construct:
```
The assert function checks whether its first argument is not false and simply returns that argument; if the argument is false (that is, false or nil), assert raises an error. Its second argument, the message, is optional, so that if you do not want to say anything in the error message, you do not have to. Beware, however, that assert is a regular function. As such, Lua always evaluates its arguments before calling the function.
```

I think that the code is correct.

Reply:
Thanks for the clarification. Looks good for me.

As an aside, if you are working on a NetworkManager library (as it seems from your examples), I'd be interested in seeing once it's published (if you intend to do so of course)